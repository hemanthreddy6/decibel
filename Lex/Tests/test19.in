distort <- (x: int): int {
    return x * x - 10;
}

customWave <- (t: float): int {
    return SIN(t) + COS(t);
}

main {
    track: audio <- load "song.wav";
    track = LOWPASS(track, 200hz) & distort;
    track1: audio <- audio(customWave, 440hz, 5s);
    track = track + track1;

    play finalTrack;
}
