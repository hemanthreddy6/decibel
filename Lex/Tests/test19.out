IDENTIFIER
<-
(
IDENTIFIER
:
INT
)
:
INT
{
RETURN
IDENTIFIER
*
IDENTIFIER
-
INT_LITERAL
;
}
IDENTIFIER
<-
(
IDENTIFIER
:
FLOAT
)
:
INT
{
RETURN
SIN
(
IDENTIFIER
)
+
COS
(
IDENTIFIER
)
;
}
MAIN
{
IDENTIFIER
:
AUDIO
<-
LOAD
STRING_LITERAL
;
IDENTIFIER
=
LOWPASS
(
IDENTIFIER
,
FLOAT_LITERAL
)
&
IDENTIFIER
;
IDENTIFIER
:
AUDIO
<-
AUDIO
(
IDENTIFIER
,
FLOAT_LITERAL
,
FLOAT_LITERAL
)
;
IDENTIFIER
=
IDENTIFIER
+
IDENTIFIER
;
PLAY
IDENTIFIER
;
}
